### PR TITLE
Added support for PHPUnit 8

### PR DESCRIPTION
**Pull request by core23**
## Subject

I am targeting this branch, because this is BC.

## Changelog

```markdown
### Added
- Added support for PHPUnit 8 for provided `BlockServiceTestCase`
```

### CODE REVIEW

**Comment by OskarStark** [comment]
Why making it internal?

**Comment by core23** [comment]
You should not extend internal classes. This hack is used to provide phpunit with and without return types.

**Comment by OskarStark** [comment]
I know but before it wasn’t internal and I use it to test my custom blocks and this was the initial goal imo. Doing the opposite is a BC break to me, at least in the stable branch.

The hack 👍🏻

File: src/Test/BlockServiceTestCase.php [hunk 1]
```diff
@@ -99,3 +104,22 @@ protected function assertSettings(array $expected, BlockContextInterface $blockC
         $this->assertSame($completeExpectedOptions, $blockSettings);
     }
 }
+
+// NEXT_MAJOR: Remove this hack when dropping support for PHPUnit 7
+if (interface_exists('PHPUnit_Framework_MockObject_MockObject')) {
+    abstract class BlockServiceTestCase extends InternalBlockServiceTestCase
```
Comment (core23) [inline src/Test/BlockServiceTestCase.php]: We add the original class name here @OskarStark

So this is still BC

**Comment by core23** [comment]
We could also do the same like symfony does: https://github.com/symfony/symfony/pull/32842

File: src/Test/BlockServiceTestCase.php [hunk 2]
```diff
@@ -99,3 +104,22 @@ protected function assertSettings(array $expected, BlockContextInterface $blockC
         $this->assertSame($completeExpectedOptions, $blockSettings);
     }
 }
+
+// NEXT_MAJOR: Remove this hack when dropping support for PHPUnit 7
+if (interface_exists('PHPUnit_Framework_MockObject_MockObject')) {
```
Comment (phansys) [inline src/Test/BlockServiceTestCase.php]: Why not `version_compare(\PHPUnit\Runner\Version::id(), '8.0', '<')`?

Comment (core23) [reply src/Test/BlockServiceTestCase.php]: Fixed

**Pull request merged**